CSharp_Course_Project/Task3_Queue/StudentQueue.cs

```diff
@@ -0,0 +1,165 @@
+
+        private readonly Queue<string> tasks = new Queue<string>();
+
+        public void ManageStudentQueue()
+        {
+            while (true)
+            {
+                Console.WriteLine($"Please, choose one of the option:{Environment.NewLine}" +
+                $"1 - Add student{Environment.NewLine}" +
+                $"2 - Remove student{Environment.NewLine}" +
+                $"3 - Print student list{Environment.NewLine}" +
+                $"4 - Manage task{Environment.NewLine}" +
+                $"0 - Escape{Environment.NewLine}");
+                var option = Console.ReadLine();
+
+                switch (option)
+                {
+                    case "1":
+                        this.AddStudent();
+                        break;
+                    case "2":
+                        this.RemoveStudent();
+                        break;
+                    case "3":
+                        this.PrintStudent();
+                        break;
+                    case "4":
+                        this.ManageTask();
+                        break;
+                    case "0":
+                        break;
+                    default:
+                        Console.WriteLine("Wrong option. Please, try again...");
+                        continue;
+                }
+
+                if (option == "0")
+                    break;
+            }
+        }
+
+        public void AddStudent()
+        {
+            while (true)
+            {
+                Console.Write("Please enter a student name: ");
+                var name = Console.ReadLine();
+
+                Console.Write("Please enter a student surname: ");
+                var surname = Console.ReadLine();
+
+                Console.Write("Please enter a student course (1-5): ");
+                var course = Byte.TryParse(Console.ReadLine(), out byte courseNumber);
+
+                Console.Write("Please enter a student group: ");
+                var group = Console.ReadLine();
+
+                Console.Write("Please enter a student faculty: ");
+                var faculty = Console.ReadLine();
+
+                var student = new Student(name, surname, courseNumber, group, faculty);
+                this.students.Add($"{student.Name} {student.Surname}", student);
+
+                Console.WriteLine("Do you want to add one more student? Y/N");
+                var addResult = Console.ReadLine();
+                if (addResult.ToLower() == "y")
+                    continue;
+                else
+                    break;
+            }
+        }
+
+        public void RemoveStudent()
+        {
+            Console.WriteLine($"Choose one of the option to remove student(s)?{Environment.NewLine}" +
+                $"1 - Clear student list{Environment.NewLine}" +
+                $"2 - Remove student by name and surname");
+            var removeStatus = Console.ReadLine();
+            if (removeStatus == "1")
```

Comment:
Please put empty line before 'if' statement to emphasize the logic.
Please use this approach in general.

Reply:
fixed